File: solutions/go/lasagna/1/lasagna.go
package lasagna

// TODO: define the 'OvenTime' constant
const OvenTime int = 40

// RemainingOvenTime returns the remaining minutes based on the `actual` minutes already in the oven.
func RemainingOvenTime(actualRemainingTime int) int {
    return OvenTime - actualRemainingTime
}

// PreparationTime calculates the time needed to prepare the lasagna based on the amount of layers.
func PreparationTime(numberOfLayers int) int {
    return 2 * numberOfLayers;
}

// ElapsedTime calculates the time elapsed cooking the lasagna. This time includes the preparation time and the time the lasagna is baking in the oven.
func ElapsedTime(numberOfLayers int, actualMinutesInOven int) int{
    return (2 * numberOfLayers) + actualMinutesInOven
}

func main(){
    RemainingOvenTime(30)
    PreparationTime(2)
    ElapsedTime(3, 20)
}

File: solutions/python/guidos-gorgeous-lasagna/1/lasagna.py
"""Functions used in preparing Guido's gorgeous lasagna.

Learn about Guido, the creator of the Python language:
https://en.wikipedia.org/wiki/Guido_van_Rossum

This module provides helper functions to calculate baking and preparation times
for a lasagna recipe.
"""

# Constants
EXPECTED_BAKE_TIME = 40        # minutes
PREPARATION_TIME = 2           # minutes per layer


def bake_time_remaining(elapsed_bake_time):
    """Calculate the bake time remaining.

    :param elapsed_bake_time: int - baking time already elapsed (in minutes).
    :return: int - remaining bake time (in minutes).

    Takes the actual minutes the lasagna has been baking and returns how many
    minutes it still needs to bake based on the EXPECTED_BAKE_TIME.
    """
    return EXPECTED_BAKE_TIME - elapsed_bake_time


def preparation_time_in_minutes(number_of_layers):
    """Calculate the preparation time.

    :param number_of_layers: int - the number of lasagna layers.
    :return: int - total preparation time (in minutes).

    Assumes each layer takes PREPARATION_TIME minutes to prepare.
    """
    return PREPARATION_TIME * number_of_layers


def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """Calculate the total elapsed cooking time.

    :param number_of_layers: int - number of layers in the lasagna.
    :param elapsed_bake_time: int - elapsed baking time (in minutes).
    :return: int - total time spent cooking (in minutes).

    Adds preparation time to the elapsed baking time to get the total.
    """
    return preparation_time_in_minutes(number_of_layers) + elapsed_bake_time
